src/pages/ProductDetail/ProductDetail.js

```diff
@@ -1,9 +1,270 @@
 import React, { Component } from 'react';
+import ProductPhoto from './ProductPhoto';
+import ProductSize from './ProductSize';
+import ProductBonus from './ProductBonus';
+import ShoppingBag from './ShoppingBag';
 import './ProductDetail.scss';
 
 class ProductDetail extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      productInfoList: [],
+      imageNum: 0,
+      isSize: true,
+      isBonus: true,
+      isShoppingBag: true,
+      size: '',
+    };
+  }
+
+  componentDidMount() {
+    fetch('/data/productData.json')
+      .then(res => res.json())
+      .then(data => {
+        this.setState({
+          productInfoList: data.results,
+        });
+      });
+  }
+
+  postProduct = () => {
+    const { productInfoList } = this.state;
+    fetch('http://10.58.7.79:8000/users/signup', {
+      method: 'POST',
+      body: JSON.stringify({
+        product_id: productInfoList.product_id,
+      }),
+    });
+  };
+
+  changeSize = size => {
+    this.setState({ size: size }, this.hideSizePick());
+  };
+
+  hideSizePick = () => {
+    const { isSize } = this.state;
+    this.setState({ isSize: !isSize });
+  };
+
+  hideBonus = () => {
+    const { isBonus } = this.state;
+    this.setState({ isBonus: !isBonus });
+  };
+
+  hideShoppingBag = () => {
+    const { isShoppingBag } = this.state;
+    this.setState({ isShoppingBag: !isShoppingBag });
+  };
+
+  changePhoto = photo => {
+    this.setState({
+      imageNum: photo,
+    });
+  };
+
+  gotoCart = () => {
+    this.props.history.push('/cart');
+  };
+
+  gotoMain = () => {
+    this.props.history.push('/mainPage');
+  };
+
   render() {
-    return <div></div>;
+    const { productInfoList, size, isSize, isBonus, isShoppingBag, imageNum } =
+      this.state;
+    return (
+      <div className="productDetail">
+        <div className="nav">임시</div>
+        <div className="list">
+          <span className="home" onClick={this.gotoMain}>
+            홈
+          </span>
+          &nbsp;&nbsp;/&nbsp;&nbsp;
+          <span className="category">
+            {productInfoList && productInfoList.section}
+          </span>
+          &nbsp; / &nbsp;
+          <span className="category">
+            {productInfoList && productInfoList.category}
+          </span>
+          &nbsp; / &nbsp;
+          {productInfoList && productInfoList.name}
+        </div>
+        <div className="productMain">
+          <div className="productMainInteraction">
+            <div className="slideBox">
+              <div className="miniPhotoBox">
+                {productInfoList.images &&
+                  productInfoList.images.map((image, idx) => {
```

Comment:
```suggestion
                {productInfoList.images?.map((image, idx) => {
```
- 한가지 의문사항은 productList의 초기 값은 빈 배열인데 사용하고 계신 형태는 객체처럼 다루고 계시네요
- 만약 초기값이 객체가 맞다면 변수명을 productList가 아니라 product나 productInfo로 바꾸시고 객체 형태에 알맞은 조건부 렌더링을 사용하시면 됩니다
- 이 경우 옵셔널 체이닝 문법을 사용하면 보다 간단하게 처리할 수 있을거 같네요

Reply:
초기 값 객체로 수정하고 옵셔널 체이닝 사용해보겠습니다